src/test/java/org/folio/rest/api/HoldingsStorageTest.java

```diff
@@ -813,6 +829,443 @@ public void cannotCreateHoldingWithoutPermanentLocation() throws Exception {
+  }
+
+  @Test
+  public void canUsePutToCreateAHoldingsWhenHRIDNotSupplied()
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+    log.info("Starting canUsePutToCreateAHoldingsWhenHRIDNotSupplied");
+
+    final UUID instanceId = UUID.randomUUID();
+
+    instancesClient.create(smallAngryPlanet(instanceId));
+
+    final UUID holdingsId = UUID.randomUUID();
+
+    setHoldingsSequence(1);
+
+    holdingsClient.replace(holdingsId, new HoldingRequestBuilder()
+      .withId(holdingsId)
+      .forInstance(instanceId)
+      .withPermanentLocation(mainLibraryLocationId)
+      .withTags(new JsonObject().put("tagList", new JsonArray().add(TAG_VALUE))));
+
+    final Response getResponse = holdingsClient.getById(holdingsId);
+
+    assertThat(getResponse.getStatusCode(), is(HttpURLConnection.HTTP_OK));
+
+    final JsonObject holdingsFromGet = getResponse.getJson();
+
+    assertThat(holdingsFromGet.getString("hrid"), is("ho00000001"));
+
+    log.info("Finished canUsePutToCreateAHoldingsWhenHRIDNotSupplied");
+  }
+
+  @Test
+  public void canUsePutToCreateAHoldingsWhenHRIDIsSupplied()
+      throws MalformedURLException,
+      InterruptedException,
+      ExecutionException,
+      TimeoutException {
+    log.info("Starting canUsePutToCreateAHoldingsWhenHRIDIsSupplied");
+
+    final UUID instanceId = UUID.randomUUID();
+
+    instancesClient.create(smallAngryPlanet(instanceId));
+
+    final UUID holdingsId = UUID.randomUUID();
+
+    final String hrid = "TEST1001";
+
+    holdingsClient.replace(holdingsId, new HoldingRequestBuilder()
+      .withId(holdingsId)
+      .forInstance(instanceId)
+      .withPermanentLocation(mainLibraryLocationId)
+      .withHrid(hrid)
+      .withTags(new JsonObject().put("tagList", new JsonArray().add(TAG_VALUE))));
+
+    final Response getResponse = holdingsClient.getById(holdingsId);
+
+    assertThat(getResponse.getStatusCode(), is(HttpURLConnection.HTTP_OK));
+
+    final JsonObject holdingsFromGet = getResponse.getJson();
+
+    assertThat(holdingsFromGet.getString("hrid"), is(hrid));
+
+    log.info("Finished canUsePutToCreateAHoldingsWhenHRIDIsSupplied");
+  }
+
+  @Test
+  public void canPostSynchronousBatchWithGeneratedHRID() {
+    log.info("Starting canPostSynchronousBatchWithGeneratedHRID");
+
+    setHoldingsSequence(1);
+
+    final JsonArray holdingsArray = threeHoldings();
+
+    assertThat(postSynchronousBatch(holdingsArray), statusCodeIs(HttpStatus.HTTP_CREATED));
+
+    holdingsArray.stream()
+        .map(o -> (JsonObject) o)
+        .forEach(holdings -> {
+          final Response response = getById(holdings.getString("id"));
+          assertExists(response, holdings);
+          assertHRIDRange(response, "ho00000001", "ho00000003");
+        });
+
+    log.info("Finished canPostSynchronousBatchWithGeneratedHRID");
+  }
+
+  @Test
+  public void canPostSynchronousBatchWithSuppliedAndHRID() {
```

Comment:
Should this be supplied and generated?

```suggestion
  public void canPostSynchronousBatchWithSuppliedAndGeneratedHRID() {
```